src/lib/select/select.ts

```diff
@@ -375,4 +425,169 @@ export class MdSelect implements AfterContentInit, ControlValueAccessor, OnDestr
+    const optionOffsetFromScrollTop = SELECT_OPTION_HEIGHT * selectedIndex;
+    const halfOptionHeight = SELECT_OPTION_HEIGHT / 2;
+
+    // Starts at the optionOffsetFromScrollTop, which scrolls the option to the top of the
+    // scroll container, then subtracts the scroll buffer to scroll the option down to
+    // the center of the overlay panel. Half the option height must be re-added to the
+    // scrollTop so the option is centered based on its middle, not its top edge.
+    const optimalScrollPosition = optionOffsetFromScrollTop - scrollBuffer + halfOptionHeight;
+    return clampValue(0, optimalScrollPosition, maxScroll);
+  }
+
+  /**
+   * Calculates the y-offset of the select's overlay panel in relation to the
+   * top start corner of the trigger. It has to be adjusted in order for the
+   * selected option to be aligned over the trigger when the panel opens.
+   */
+  private _calculateOverlayOffset(selectedIndex: number, scrollBuffer: number,
+                                  maxScroll: number): number {
+    let optionOffsetFromPanelTop: number;
+
+    if (this._scrollTop === 0) {
+      optionOffsetFromPanelTop = selectedIndex * SELECT_OPTION_HEIGHT;
+    } else if (this._scrollTop === maxScroll) {
+      const firstDisplayedIndex = this.options.length - SELECT_MAX_OPTIONS_DISPLAYED;
+      const selectedDisplayIndex = selectedIndex - firstDisplayedIndex;
+
+      // Because the panel height is longer than the height of the options alone,
+      // there is always extra padding at the top or bottom of the panel. When
+      // scrolled to the very bottom, this padding is at the top of the panel and
+      // must be added to the offset.
+      optionOffsetFromPanelTop =
+          selectedDisplayIndex * SELECT_OPTION_HEIGHT + SELECT_PANEL_PADDING_Y;
+    } else {
+      // If the option was scrolled to the middle of the panel using a scroll buffer,
+      // its offset will be the scroll buffer minus the half height that was added to
+      // center it.
+      optionOffsetFromPanelTop = scrollBuffer - SELECT_OPTION_HEIGHT / 2;
+    }
+    // Must adjust for the difference in height between the option and the trigger,
+    // so the text will align on the y axis.
+    const heightDiff = (SELECT_OPTION_HEIGHT - SELECT_TRIGGER_HEIGHT) / 2;
+
+    // The final offset is the option's offset from the top, adjusted for the height
+    // difference, multiplied by -1 to ensure that the overlay moves in the correct
+    // direction up the page.
+    return optionOffsetFromPanelTop * -1 - heightDiff;
+  }
+
+  /**
+   * Checks that the attempted overlay position will fit within the viewport.
+   * If it will not fit, tries to adjust the scroll position and the associated
+   * y-offset so the panel can open fully on-screen. If it still won't fit,
+   * sets the offset back to 0 to allow the fallback position to take over.
+   */
+  private _checkOverlayWithinViewport(maxScroll: number): void {
```

Comment:
Why isn't the behavior
"default position" > "fallback position" ( * n) > slide to stay on-screen

With the last one being part of the connected position strategy?